securedrop/journalist_app/admin.py

```diff
@@ -288,16 +288,22 @@ def delete_user(user_id: int) -> werkzeug.Response:
             current_app.logger.error(
                 "Admin {} tried to delete itself".format(g.user.username))
             abort(403)
-        elif user:
-            db.session.delete(user)
-            db.session.commit()
-            flash(gettext("Deleted user '{user}'.").format(
-                user=user.username), "notification")
-        else:
+        elif not user:
             current_app.logger.error(
                 "Admin {} tried to delete nonexistent user with pk={}".format(
                     g.user.username, user_id))
             abort(404)
+        elif user.username == "deleted":
+            # Do not flash because the interface does not expose this.
```

Comment:
I would recommend using an explicit field on the Journalist model to identify the deleted user, such as 'is_special_deleted_journalist: bool' (and it can have some kind of unique constraint).

Using the name might cause problems later: what if an instance (somehow) ever has a journalist named "deleted", etc.

Reply:
SecureDrop already prevents creating a user with the name "deleted" so I don't think there's any worry with collisions. I don't see much value in having a separate column for it either, but I do like the suggestion to have a specific method for it so we're not hardcoding the username everywhere

```python
def is_deleted_user(self) -> bool:
    return self.username == "deleted"
```